pandas/core/internals.py

```diff
@@ -874,22 +875,7 @@ def setitem(self, indexer, value, mgr=None):
         values = transf(values)
 
         # length checking
-        # boolean with truth values == len of the value is ok too
```

Comment:
This was refactored out to `pandas/util/_validators` since I needed it in `ExtensionArray.setitem`.